pandas/core/arrays/datetimelike.py

```diff
@@ -2111,7 +2111,15 @@ def factorize(
                 codes = codes[::-1]
                 uniques = uniques[::-1]
             return codes, uniques
-        # FIXME: shouldn't get here; we are ignoring sort
+
+        if sort:
+            # algorithms.factorize only passes sort=True here when freq is not None,
+            #  so this should not be reached.
+            raise NotImplementedError(
+                f"The 'sort' keyword in {type(self).__name__}.factorize is "
+                "ignored unless arr.freq is not None. To factorize with sort, "
+                "call pd.factorize(obj, sort=True) instead."
+            )
```

Comment:
Add a test for this in `pandas/tests/indexes/datetimes/methods/test_factorize.py`?

Reply:
decided to revert this for now